### PR TITLE
print actx type on startup

### PR DESCRIPTION
**Pull request by matthiasdiener**
**Questions for the review**:
- [ ] Is the scope and purpose of the PR clear?
  - [ ] The PR should have a description.
  - [ ] The PR should have a guide if needed (e.g., an ordering).
- [ ] Is every top-level method and class documented? Are things that should be documented actually so?
- [ ] Is the interface understandable? (I.e. can someone figure out what stuff does?) Is it well-defined?
- [ ] Does the implementation do what the docstring claims?
- [ ] Is everything that is implemented covered by tests?
- [ ] Do you see any immediate risks or performance disadvantages with the design? Example: what do interface normals attach to?

### CODE REVIEW

File: mirgecom/mpi.py [hunk 1]
```diff
@@ -121,6 +121,20 @@ def wrapped_func(*args, **kwargs):
...
+            from meshmode.array_context import PyOpenCLArrayContext as OldEagerActx
+            if actx_class == OldEagerActx:
+                print(
+                    f"*** Eager mode with deprecated '{actx_class}. ***")
+            elif actx_class == NewEagerActx:
+                print(
+                    f"*** Eager mode with '{actx_class}. ***")
+            else:
+                print(
+                    f"*** Lazy mode with '{actx_class}. ***")
```
Comment (inducer) [inline mirgecom/mpi.py:136]: I think this ought to use `logging`.

File: mirgecom/mpi.py [hunk 2]
```diff
@@ -121,6 +121,20 @@ def wrapped_func(*args, **kwargs):
             raise RuntimeError("mpi4py.MPI imported before designated MPI entry "
                         "point. Check for prior imports.")
 
+        actx_class = kwargs.pop("actx_class", None)
+        if actx_class:
```
Comment (inducer) [inline mirgecom/mpi.py:125]: I'm not sure about this being mixed in with the MPI functionality. I can see *why*, but I'm not sure it's a good idea.

**Comment by matthiasdiener** [comment]
https://github.com/inducer/grudge/pull/230 implements something similar in grudge.

**Comment by matthiasdiener** [comment]
Closing this since https://github.com/inducer/grudge/pull/230 has been merged.

**Pull request closed without merging**